File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        
        if(nums.length == 0)
            return 0;
    
        
        Arrays.sort(nums);
        int result = 1;
        int maxResult = 1;
        
        for(int i=0;i<nums.length-1;i++) {
            if(nums[i] != nums[i+1]) { 
            int currentELement = nums[i];
            int nextElement = nums[i+1];
            if(currentELement + 1 == nextElement) {
                result = result + 1;
                maxResult = Math.max(maxResult, result);
                
            } else {                
                
                result = 1;
            }
          }
        }
     return maxResult;   
    }
}

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    let s1= new Set(nums);
    let longest =0;
    for(let n of nums){
        if(s1.has(n-1)==false){
            let len = 0;
            while(s1.has(n+len)){
                len+=1;
            }
            longest=longest>len?longest:len;
        }
    }
    
    return longest;
};